libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationUIBinders.kt

```diff
@@ -5,12 +5,44 @@ import com.mapbox.navigation.dropin.binder.infopanel.InfoPanelTripProgressBinder
 import com.mapbox.navigation.dropin.component.maneuver.ManeuverViewBinder
 import com.mapbox.navigation.dropin.component.roadlabel.RoadNameViewBinder
 import com.mapbox.navigation.dropin.component.speedlimit.SpeedLimitViewBinder
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
 
-class NavigationUIBinders(
-    val speedLimit: UIBinder = SpeedLimitViewBinder(),
-    val maneuver: UIBinder = ManeuverViewBinder(),
-    val roadName: UIBinder = RoadNameViewBinder(),
-    val infoPanelTripProgressBinder: UIBinder = InfoPanelTripProgressBinder(),
-    val infoPanelHeaderBinder: UIBinder? = null,
-    val infoPanelContentBinder: UIBinder? = null
-)
+internal class NavigationUIBinders {
```

Comment:
I think we should be able to add generic options to these. For example, the developer will expect the route options to update if they send a `Customization` with new route options.

Could this be `NavigationCustomizations` or `UICusomizations`?

Reply:
🤷‍♂️

Reply:
It does matter though, we should either have `UIBinderCustomization` and `UIOptionCustomization`

If we have a single `Customization` it would be good to have them behave in a similar way. It will also be helpful for when we want to add a customization field. 

For example, if I try to add a new `Customization`; it would be good to know i should also add to this class

Reply:
🤷‍♂️ No idea. If you have better idea how to do this go ahead and change it. All commits are welcome.

Reply:
I would look at it from the opposite spectrum. This PR shouldn't have anything to do with customizing routeLine and routeArrow options. I would revert those changes and let this class be as is so it's clear that this is only for view binders.